src/Processors/Transforms/JoiningTransform.cpp

```diff
@@ -387,11 +390,21 @@ void DelayedJoinedBlocksWorkerTransform::work()
     if (!task)
         return;
 
-    Block block = task->delayed_blocks->next();
-
+    Block block;
+    /// All joined and non-joined rows from left stream are emitted, only right non-joined rows are left
+    if (!task->delayed_blocks->isFinished())
```

Comment:
```suggestion
    if (!task->finished)
```

Reply:
I remove `DelayedBlocksTask::finish`. `DelayedBlocksTask::finish` is used as a sign of all buckets are finished.  But use `DelayedBlocksTask::delayed_blocks == nullptr` instead.